### PR TITLE
Make adminJoin fail harder.

### PR DESCRIPTION
**Pull request by Raynos**
It currently only warns when it should really fail.

reviewers: @jwolski

cc: @sh1mmer

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -197,7 +197,7 @@ RingPop.prototype.bootstrap = function bootstrap(bootstrapFile, callback) {
                 err: err.message,
                 address: self.hostPort
             });
-            if (callback) callback(new Error(failedMsg));
+            if (callback) callback(err);
```
Comment (jwolski) [inline index.js]: can you inline `failedMsg` now?

File: test/integration/admin_join_test.js [hunk 2]
```diff
@@ -0,0 +1,19 @@
+'use strict';
```
Comment (jwolski) [inline test/integration/admin_join_test.js]: COPYRIGHT

File: lib/swim.js [hunk 3]
```diff
@@ -17,6 +17,15 @@
 // LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 // OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
 // THE SOFTWARE.
+var TypedError = require('error/typed');
+
+var NoHostsError = TypedError({
+    message: 'Could not find any hosts in bootstrap to join.\n' +
+        'Try adding more bootstrapHosts.\n',
+    type: 'ringpop.swim.no-hosts',
+    bootstrapHosts: null
```
Comment (jwolski) [inline lib/swim.js]: did you intend to parameterize {boostrapHosts} within the error message?

Comment (Raynos) [inline lib/swim.js]: was not planning to. This is an example of meta data not in the message.

Comment (jwolski) [inline lib/swim.js]: cool

**Comment by Raynos** [comment]
Addressed review feedback.

File: package.json [hunk 4]
```diff
@@ -14,7 +14,9 @@
     "jshint": "jshint --verbose *.js",
     "cover": "istanbul cover --print detail --report html test/index.js",
     "view-cover": "opn coverage/index.html",
-    "test": "npm run jshint && node test/index.js"
+    "test": "npm run jshint && node test/index.js",
+    "check-licence": "uber-licence --dry",
+    "add-licence": "uber-licence"
```
Comment (jwolski) [inline package.json]: alphabetize

Comment (Raynos) [inline package.json]: done.

**Comment by Raynos** [comment]
@jwolski updated.

I alphabetized, except test, test is at the top as its the most important one.

**Comment by jwolski** [comment]
LGTM

**Comment by Raynos** [comment]
@jwolski this is a major change. Code that used to not error, will now error.

**Pull request merged**